### DISCUSSION TITLE
Match all files under `.circleci` directory for CircleCI manager

### DISCUSSION THREAD

**Discussion by mkniewallner** [renovatebot/renovate#27807] [Suggest an Idea]
### Tell us more.

CircleCI supports [dynamic configuration](https://circleci.com/docs/using-dynamic-configuration/), which allows spreading the workflow definition files across multiple YAML files, instead of just `.circleci/config.yml`.

As can be seen in https://circleci.com/docs/using-dynamic-configuration/#execute-specific-workflows-or-steps-based-on-which-files-are-modified, 2 files are used as an example, both having dependencies that Renovate is able to handle (CircleCI orbs).

The default regex for `fileMatch` currently only targets `.circleci/config.yml` and `.circleci/config.yaml`:
https://github.com/renovatebot/renovate/blob/049c59c09667722667df0bf1a703a971cafa997b/lib/modules/manager/circleci/index.ts#L13

Since `.circleci` directory is likely to only contain CircleCI-specific YAML files, would it make sense to have a broader regex that targets all YAML files in this directory by default? Something like `(^|/)\\.circleci/\\.+\\.ya?ml$`?

**Comment by rarkins** [comment]
PR welcome

**Comment by mkniewallner** [comment]
Resolved by https://github.com/renovatebot/renovate/pull/27809.
